libsplinter/src/peer/mod.rs

```diff
@@ -343,20 +349,21 @@ impl PeerManager {
                 ))
             })?;
 
-        let shutdown_handle = ShutdownHandle {
+        let shutdown_signaler = ShutdownSignaler {
             sender: sender.clone(),
             pacemaker_shutdown_signaler,
         };
 
         Ok(PeerManager {
-            shutdown_handle,
+            shutdown_signaler,
             join_handle,
             sender,
         })
     }
 }
 
-/// Handles sending a shutdown message to the `PeerManager` and the `Pacemaker`
+/// Handles sending a shutdown message to the `PeerManager` and the `Pacemaker`.
+/// Deprecated, use ShutdownSignaler instead.
```

Comment:
Can the `deprecated` attribute be added to a struct as well?

Reply:
It can be, but then it always raises the deprecation warning because of the Impl then uses the struct.